pydoctor/epydoc2stan.py

```diff
@@ -722,27 +722,28 @@ def format_summary(obj: model.Documentable) -> Tag:
 def format_undocumented(obj: model.Documentable) -> Tag:
     """Generate an HTML representation for an object lacking a docstring."""
 
-    subdocstrings: DefaultDict[str, int] = defaultdict(int)
-    subcounts: DefaultDict[str, int]  = defaultdict(int)
-    for subob in obj.contents.values():
-        k = subob.kind.lower()
-        subcounts[k] += 1
-        if subob.docstring is not None:
-            subdocstrings[k] += 1
+    sub_objects_with_docstring_count: DefaultDict[model.KindClass, int] = defaultdict(int)
+    sub_objects_total_count: DefaultDict[model.KindClass, int]  = defaultdict(int)
+    for sub_ob in obj.contents.values():
+        k = sub_ob.kind
+        sub_objects_total_count[k] += 1
+        if sub_ob.docstring is not None:
+            sub_objects_with_docstring_count[k] += 1
     if isinstance(obj, model.Package):
-        subcounts['module'] -= 1
+        sub_objects_total_count[model.KindClass.MODULE] -= 1
 
     tag: Tag = tags.span(class_='undocumented')
-    if subdocstrings:
-        plurals = {'class': 'classes', 'property': 'properties'}
+    if sub_objects_with_docstring_count:
+        
         kind = obj.kind
         assert kind is not None # if kind is None, object is invisible
         tag(
-            "No ", kind.lower(), " docstring; ",
+            "No ", format_kind(kind), " docstring; ",
             ', '.join(
-                f"{subdocstrings[k]}/{subcounts[k]} "
-                f"{plurals.get(k, k + 's')}"
-                for k in sorted(subcounts)
+                f"{sub_objects_with_docstring_count[k]}/{sub_objects_total_count[k]} "
+                f"{format_kind(k, plural=sub_objects_with_docstring_count[k]>=2).lower()}"
```

Comment:
Maybe `format_kind()` should return the description in lower case and we can call `.title()` when we want title case in the output.

Reply:
I think the opposite actually, lower() function drops casing information, where title() adds cases where it thinks it the best. What is there a adverb I the kind and we don’t want to upper case it ?

Reply:
That's true. While we currently don't have such word like "of" in the descriptions, it's likely we'll get more kinds in the future and they might have. So it's fine to keep it like this.